test/qunit/model/FHIRModel.integration.js

```diff
@@ -849,4 +850,28 @@ sap.ui.define([
 		this.oFhirModel.submitChanges("transaction", fnSuccessCallback);
 	});
 
+	QUnit.test("Test remove items from list and verify the contexts before and after submitting the changes", function (assert) {
+		var oListBinding = this.oFhirModel.bindList("/Practitioner", undefined, undefined, undefined, { groupId: "patientDetails" });
+		this.oFhirModel.aBindings.push(oListBinding);
+		var done = assert.async();
+		var fnAssertion = function () {
+			var iCurrentLength = oListBinding.getContexts().length;
+			var aContext = oListBinding.getContexts();
+			var sResPath = aContext[0].sPath;
+			this.oFhirModel.remove([sResPath], undefined, "patientDetails");
+			assert.deepEqual(oListBinding.getContexts().length, iCurrentLength - 1, "List doesn't show the removed items even if the changes are not submitted");
+			this.oFhirModel.resetChanges("patientDetails");
+			assert.deepEqual(oListBinding.getContexts().length, iCurrentLength, "List shows the previously removed items if the changes are not submitted and reset changes is triggered");
+			assert.deepEqual(this.oFhirModel.mRemovedResources["Practitioner"], undefined, "Removed Resources of the model is correctly cleared during reset changes");
```

Comment:
check here if the object key `Practitioner` still exists on `Practitioner` via `this.oFhirModel.mRemovedResources.hasOwnProperty`. If this is missing the `delete` operation is untested

Reply:
also below when you perform the checks after submitting

Reply:
done